### PR TITLE
Parse GraphQL in external files

### PR DESCRIPTION
**Pull request by dfreeman**
Coming from the conversation in #25, here's an implementation that precompiles `.graphql` files to JS at build time, translating `#import '<path>'` directives into ES6 `import '<path>';` in the output. Relative or absolute import paths should behave exactly the same as they would in a regular JS module in the same location.

The primary question I have looking at this is what a reasonable "Embery" project structure would be for these files. The implementation itself doesn't care, but it seems like the dummy app and documentation ought to set a good example (whatever we believe that to be). Regarding documentation, I haven't updated the README on this branch yet—hoping to get thoughts on reasonable conventions first :)

### CODE REVIEW

**Comment by viniciussbs** [comment]
The content of a `.graphql` file is a [Query Document](https://facebook.github.io/graphql/#sec-Language.Query-Document). So, I propose:

```
app/query-documents/fragments
app/query-documents/mutations
app/query-documents/queries
app/query-documents/subscriptions
```

**Comment by n1ru4l** [comment]
@viniciussbs https://github.com/emberjs/rfcs/blob/master/text/0143-module-unification.md

**Comment by viniciussbs** [comment]
@n1ru4l What do you suggest?

**Comment by n1ru4l** [comment]
I think query-documents is not specific enough (there are queries, mutations and also subscriptions). How about `graphql-docs` or `graphql-documents`? I think the link I posted earlier is not relevant yet, since I don't know how long it will actually take until this will apply to our ember applications

**Comment by viniciussbs** [comment]
I agree with you, @n1ru4l, that `graphql-documents` is more specific. It's [how Apollo calls them](https://dev-blog.apollodata.com/the-anatomy-of-a-graphql-query-6dffa9e9e747).

```
app/graphql-documents/fragments
app/graphql-documents/mutations
app/graphql-documents/queries
app/graphql-documents/subscriptions
```

**Comment by bgentry** [comment]
I love a good bike shed :joy:

I'm indifferent to most of these, though they do seem pretty long. A possible shorter alternative is `app/gql`.

**Comment by dfreeman** [comment]
> I love a good bike shed 😂

🎉 

> they do seem pretty long. A possible shorter alternative is `app/gql`.

I lean the same direction. I like `gql` as a terse base directory (a "group" in Module Unification parlance, peer to `ui`, `init` and `data`), containing collections for the different document types. Also nice that this layout seems to fit nicely into both the current and future vision for project structure.

**Comment by dfreeman** [comment]
Updated the README and the location of the GraphQL documents in the dummy app. A few additional thoughts (probably stuff that can happen in follow-on PRs):
 - Providing blueprints to produce skeletons for each type of document would be cool, e.g. `ember g mutation create-review`
 - Now that I'm looking closely at the examples, it's possible they could use a little TLC. I'm not `fragment ReviewFragment on Human` actually makes sense 😜 
 - Not sure if it's worth documenting the tradeoff between ahead-of-time parsing in `.graphql` files vs runtime `graphql-tag` usage, but in doing this work I noticed it basically comes down to payload size vs. extra runtime work.

File: README.md [hunk 1]
```diff
@@ -99,16 +102,31 @@ same model(s).
...
+`app/gql/fragments/review-fragment.graphql`
+```graphql
+fragment ReviewFragment on Human {
+  stars
+  commentary
+}
+```
 
-const ReviewFragment = gql`
-  fragment ReviewFragment on Human {
```
Comment (bgentry) [inline README.md]: lol, no idea why I did this :shrug:

**Comment by bgentry** [comment]
You rock @dfreeman :metal: thanks!

**Pull request merged**